src/v/cluster/types.cc

```diff
@@ -293,24 +318,188 @@ std::ostream& operator<<(
 } // namespace cluster
 
 namespace reflection {
+
+// These enum encoding/decoding functions should be move into the different
+// translation unit eventually
+
+inline void read_nested(
+  iobuf_parser& in, model::compression& el, size_t const bytes_left_limit) {
+    serde::read_enum(in, el, bytes_left_limit);
+}
+
+inline void write(iobuf& out, model::compression el) {
+    serde::write_enum(out, el);
+}
```

Comment:
As the first user of serde we have to figure out answers to questions like: where serialization for supporting types go. My opinion is that serialization for a particular type should go as close as possible in the code to the thing it is serializing. Into the type as a member, if possible.

@felixguendling what do you think? where should we be locating this so we set ourselves up for scaling and making things manageable and reducing the chances of error?

Reply:
I think that it's important to keep things together that belong together - i.e. struct and the serialization of the struct. The closest you can get is by implementing it in the struct as a global `friend` function. This way you safe time if you want to read the code: if it's not inside the struct definition itself, there is no read/write function implemented - no need to search the whole code base. If you want to implement it: just add it in the struct (maybe as a convention below ctor/dtor and operators, next to other I/O like `operator<<`).

If compile times are a concern, the implementation can go into the source file. The signature of read/write functions only require `iobuf` and `iobuf_parser` which is probably already included in most places.

Additionally, I tried to simplify the code for enums with this PR: #2523
I think it will help to keep the code concise. Any feedback is welcome.